includes/Commerce/Orders.php

```diff
@@ -298,12 +299,82 @@ public function fulfill_order( \WC_Order $order, $tracking_number, $carrier ) {
 	 * @since 2.1.0-dev.1
 	 *
 	 * @param \WC_Order_Refund $refund order refund object
-	 * @param mixed $args
+	 * @param string $reason_code refund reason code
 	 * @throws SV_WC_Plugin_Exception
 	 */
-	public function add_order_refund( \WC_Order_Refund $refund, $args ) {
+	public function add_order_refund( \WC_Order_Refund $refund, $reason_code ) {
 
-		// TODO: implement
+		$plugin = facebook_for_woocommerce();
+
+		$api = $plugin->get_api( $plugin->get_connection_handler()->get_page_access_token() );
+
+		$valid_reason_codes = [
+			Refund_Request::REASON_BUYERS_REMORSE,
+			Refund_Request::REASON_DAMAGED_GOODS,
+			Refund_Request::REASON_NOT_AS_DESCRIBED,
+			Refund_Request::REASON_QUALITY_ISSUE,
+			Refund_Request::REASON_OTHER,
+			Refund_Request::REASON_WRONG_ITEM,
+		];
+
+		if ( ! in_array( $reason_code, $valid_reason_codes, true ) ) {
+			$reason_code = Refund_Request::REASON_OTHER;
+		}
+
+		try {
+
+			$parent_order = wc_get_order( $refund->get_parent_id() );
+
+			if ( ! $parent_order instanceof \WC_Order ) {
+				throw new SV_WC_Plugin_Exception( __( 'Parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$remote_id = $parent_order->get_meta( self::REMOTE_ID_META_KEY );
+
+			if ( ! $remote_id ) {
+				throw new SV_WC_Plugin_Exception( __( 'Remote ID for parent order not found.', 'facebook-for-woocommerce' ) );
+			}
+
+			$items = [];
+
+			/** @var \WC_Order_Item_Product $item */
+			foreach ( $refund->get_items() as $item ) {
```

Comment:
Per the docs:

>For partial refund, specify the item level breakdown. Not required for full refund.

We should probably only include items if the refund is partial

Reply:
Ok, what is the easiest way to know if it is partial? Comparing the total with the parent order's total?

Reply:
I think so. There's prior art in our plugins, like this [AvaTax example](https://github.com/skyverge/wc-plugins/blob/master/woocommerce-avatax/includes/class-wc-avatax-ajax.php#L396)

Reply:
Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1525/commits/29661eb37e27f8f5f8c639e02a5650720d1415d4.